presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -169,6 +171,38 @@ public void testTimestamps()
 
         assertQueryUsingH2Cte("SELECT commit_timestamp, ship_timestamp, orderkey FROM lineitem_ex WHERE year(commit_timestamp) > 1993 and year(ship_timestamp) > 1993 and year(ship_timestamp) - year(commit_timestamp) = 1");
     }
+    @Test
+    public void testDouble()
```

Comment:
It looks like `compactValues` code path is not tested. Overall, we need to test the following cases:

- read all or selected positions
- no filter or with filter
- project data out or filter only
- apply filter function (e.g. call getBlockView and call getBlock with a subset of positions)

assertFilterProject helps cover "project data out or filter only", hence, I recommend using it.

Here are some examples of the queries and coverage they provide.

assertFilterProject("discount = 4", "tax"); // tax reader reads selected positions with no filter; discount reader reads all positions with filter

assertFilterProject("discount > tax", "tax"); // discount and tax readers read all positions with no filter; getBlockView is called for discount and tax; getBlock is called with a subset of positions

assertFilterProject("discount > tax and tax < 0.01", "tax"); // tax reader reads all with filter; discount reader reads selected positions with no filter; getBlockView is called for discount and tax; compactValues is called for discount; getBlock is called with a subset of positions

Reply:
Trying to find a way to trigger the compact values code path. 

For example: 
```
assertFilterProject("discount < tax AND (extendedprice > 1500 * quantity OR discount * 100 > quantity)", "extendedprice");
```

Doesn't trigger `compactValues`. 
```
SELECT count(*) 
FROM  tpch.tiny.lineitem
WHERE 
discount < tax AND (extendedprice > 1500 * quantity OR discount * 100 > quantity)
```
returns 9121 rows, so you'd think compacting would be used. Help!

Similarly, `readAllNulls` is never called. I imagine because we never write test data that has a `presentStream` and no `dataStream`?